src/pull.go

```diff
@@ -73,6 +76,10 @@ func PullWithGitImpl(ctx context.Context, sourceURL, cacheDir string, repoName s
 	if !gitimpl.RepositoryExists(dst) {
 		fmt.Fprintf(os.Stdout, "pulling %s to %s ...\n", originRepoName, dst)
 		_, err := gitimpl.CloneRepository(dst, &git.CloneOptions{
+			Auth: &http.BasicAuth{
```

Comment:
Since the intent of the new flags is optional, `sourceUsername` and `sourceToken` could be empty here. In that case we wouldn't want to set the Auth field. A test for this behavior would be great as well.

Reply:
changed function to accept an `http.AuthMethod` interface